app/models/course/assessment.rb

```diff
@@ -131,8 +131,13 @@ def validate_presence_of_questions
   end
 
   def validate_only_autograded_questions
-    return if questions.all?(&:auto_gradable?)
-    errors.add(:base, :autograded)
+    return unless published?
```

Comment:
Now there's already a `if: ` in the validation declaration. `  validate :validate_only_autograded_questions, if: :autograded?
` 

Think I will swap the autograded and published then.